File: Q3_Cpp/src/Product/Clothing/Clothing.cpp
#include "../../../include/Product/Clothing/Clothing.h"

// Constructor for Clothing
Clothing::Clothing(std::string id, std::string name, Supplier supplier, int packagingUnits, double price, std::string material, std::string color)
    : Product(id, name, supplier, packagingUnits, price), material(material), color(color) {
}

// Getter and setter for material
std::string Clothing::getMaterial() const {
    return material;
}

void Clothing::setMaterial(const std::string& material) {
    this->material = material;
}

// Getter and setter for color
std::string Clothing::getColor() const {
    return color;
}

void Clothing::setColor(const std::string& color) {
    this->color = color;
}

File: Q1_Java/src/main/java/Product/Clothing/Clothing.java
package Product.Clothing; // Package for Clothing

import Product.Product;
import Product.Supplier;

// Abstract Clothing class, extends Product
public abstract class Clothing extends Product {
    private String material; // Material of the clothing
    private String color; // Color of the clothing

    // Constructor for Clothing
    public Clothing(String id, String name, Supplier supplier, int packagingUnits, double price, String material, String color) {
        super(id, name, supplier, packagingUnits, price); // Calling Product's constructor
        this.material = material; // Setting the material
        this.color = color; // Setting the color
    }

    // Getter and setter for material
    public String getMaterial() { return material; }
    public void setMaterial(String material) { this.material = material; }

    // Getter and setter for color
    public String getColor() { return color; }
    public void setColor(String color) { this.color = color; }
}
